libraries/standard/http/src/http_client.c

```diff
@@ -316,29 +336,45 @@ static HTTPStatus_t addHeader( HTTPRequestHeaders_t * pRequestHeaders,
                HTTP_HEADER_LINE_SEPARATOR_LEN +
                HTTP_HEADER_LINE_SEPARATOR_LEN;
 
-    /* If we have enough room for the new header line, then write it to the header buffer. */
+    /* If we have enough room for the new header line, then write it to the
+     * header buffer. */
     if( ( backtrackHeaderLen + toAddLen ) <= pRequestHeaders->bufferLen )
     {
         /* Write "<Field>: <Value> \r\n" to the headers buffer. */
 
         /* Copy the header name into the buffer. */
-        memcpy( pBufferCur, pField, fieldLen );
+
+        /* MISRA rule 21.15 flags the memcpy of two different types, const uint8_t*
+         * and const char*. These types are the same size, so this comparision is
+         * acceptable. */
+        /* coverity[misra_c_2012_rule_21_15_violation] */
+        ( void ) memcpy( pBufferCur, pField, fieldLen );
         pBufferCur += fieldLen;
 
         /* Copy the field separator, ": ", into the buffer. */
-        memcpy( pBufferCur,
-                ( const uint8_t * ) HTTP_HEADER_FIELD_SEPARATOR,
-                HTTP_HEADER_FIELD_SEPARATOR_LEN );
+
+        /* MISRA rule 21.15 flags the memcpy of two different types, const uint8_t*
```

Comment:
An explicit cast is not enough?

Reply:
No it is not because then I get a different warning about casting from signed to unsigned.

Reply:
Okay, thanks.